sdk/ai/Azure.AI.Agents/samples/Sample8_FileSearch.md

```diff
@@ -0,0 +1,160 @@
+# Sample file search with agent in Azure.AI.Agents.
+
+In this example we will create the local file, upload it to Azure and will use it in the newly created `VectorStore` for file search.
+
+1. First, we need to create agent client and read the environment variables, which will be used in the next steps.
+
+```C# Snippet:Sample_CreateAgentClient_FileSearch
+var projectEndpoint = System.Environment.GetEnvironmentVariable("PROJECT_ENDPOINT");
+var modelDeploymentName = System.Environment.GetEnvironmentVariable("MODEL_DEPLOYMENT_NAME");
+AgentsClient client = new(endpoint: new Uri(projectEndpoint), tokenProvider: new DefaultAzureCredential());
+OpenAIClient openAIClient = client.GetOpenAIClient();
+```
+
+2. We will create a toy example file and upload it using OpenAI mechanism.
+
+Synchronous sample: 
+```C# Snippet:Sample_UploadFile_FileSearch_Sync
+string filePath = "sample_file_for_upload.txt";
+System.IO.File.WriteAllText(
+    path: filePath,
+    contents: "The word 'apple' uses the code 442345, while the word 'banana' uses the code 673457.");
+OpenAIFileClient fileClient = openAIClient.GetOpenAIFileClient();
+OpenAIFile uploadedFile = fileClient.UploadFile(filePath: filePath, purpose: FileUploadPurpose.Assistants);
+System.IO.File.Delete(filePath);
+```
+
+Asynchronous sample:
+```C# Snippet:Sample_UploadFile_FileSearch_Async
+string filePath = "sample_file_for_upload.txt";
+System.IO.File.WriteAllText(
+    path: filePath,
+    contents: "The word 'apple' uses the code 442345, while the word 'banana' uses the code 673457.");
+OpenAIFileClient fileClient = openAIClient.GetOpenAIFileClient();
+OpenAIFile uploadedFile = await fileClient.UploadFileAsync(filePath: filePath, purpose: FileUploadPurpose.Assistants);
+System.IO.File.Delete(filePath);
+```
+
+3. Create the `VectorStore` and provide it with uploaded file ID.
+
+Synchronous sample:
+```C# Snippet:Sample_CreateVectorStore_FileSearch_Sync
+VectorStoreClient vctStoreClient = openAIClient.GetVectorStoreClient();
+VectorStoreCreationOptions vctOptions = new()
+{
+    Name = "MySampleStore",
+    FileIds = { uploadedFile.Id }
+};
+VectorStore vectorStore = vctStoreClient.CreateVectorStore(options: vctOptions);
+```
+
+Asynchronous sample:
+```C# Snippet:Sample_CreateVectorStore_FileSearch_Async
+VectorStoreClient vctStoreClient = openAIClient.GetVectorStoreClient();
+VectorStoreCreationOptions vctOptions = new()
+{
+    Name = "MySampleStore",
+    FileIds = { uploadedFile.Id }
+};
+VectorStore vectorStore = await vctStoreClient.CreateVectorStoreAsync(
+    new VectorStoreCreationOptions()
+);
+```
+
+2. Now we can create an agent capable of using File search. 
+
+Synchronous sample:
+```C# Snippet:Sample_CreateAgent_FileSearch_Sync
+PromptAgentDefinition agentDefinition = new(model: modelDeploymentName)
+{
+    Instructions = "You are a helpful agent that can help fetch data from files you know about.",
+    Tools = {
```

Comment:
With the exception of trivial one-item initializer lists, it's generally good to consistently follow Allman brace style per [conventions](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions#style-guidelines); 1TBS can be elegant and is standard in a lot of other cases, but .NET generally loves having its braces on their own lines.